cmd/migrate-snapshots/writer.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	v1beta1snapshot "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	fileVolumeSnapshotClassDeletionPolicy = "volumesnapshotclass_deletion-policy"
+	subDirectoryVolumeSnapshotContents    = "01-volumesnapshotcontents"
+	subDirectoryVolumeSnapshots           = "02-volumesnapshots"
+)
+
+type objectBuffer struct {
+	classDeletionPolicy string
+	volSnapshotContents []v1beta1snapshot.VolumeSnapshotContent
+	volSnapshots        []v1beta1snapshot.VolumeSnapshot
+}
+
+type volumeSnapshotWriter struct {
+	directory string
+	objBuffer objectBuffer
+}
+
+func newVolumeSnapshotWriter(directory string) (*volumeSnapshotWriter, error) {
+	if directory != "" {
+		var err error
+		directory, err = filepath.Abs(directory)
```

Comment:
Might be worth also checking that the directory is a directory if it exists (via `os.Stat`). If it's a regular file the `MkdirAll` below will fail.

Reply:
That's a good idea. Will do.